### ISSUE TITLE
CrossValidation with class weights causes error

### ISSUE THREAD

**Issue by sh0tcall3r** [catboost/catboost#2129]
Problem:
Hi, I faced couple of issues during learning catboost classifier in multiclass case. 
1) I get the following error: 
C:/Program Files (x86)/Go Agent/pipelines/BuildMaster/catboost.git/catboost/libs/metrics/metric.cpp:6283: If non-default weights for objects are not set, the 'use_weights' parameter must not be specified 
when I set my catboost classifier like that:
model = CatBoostClassifier(
                    custom_loss=[metrics.Accuracy(), metrics.AUC(use_weights=True)],
                    loss_function='MultiClass',
                    class_weights=class_weights,
                    nan_mode="Max",
                    verbose=False
                    )
and cv like this:
cv_data = cv(
    data_pool,
    cv_params,
    stratified=True,
    early_stopping_rounds=50,
    plot=True
)

2) And this error:
C:/Program Files (x86)/Go Agent/pipelines/BuildMaster/catboost.git/catboost/libs/train_lib/cross_validation.cpp:92: ClassLabels in dataprocessing options and in training data must match
when I remove AUC from custom loss metric in the classifier.

This's quite frustrating, I cannot resolve it by myself. Please help.

catboost version: 1.0.6
Operating System: Windows 10 x64
CPU: AMD Ryzen 7 5800H
GPU:

**Comment by Adelegg** [comment]
Hello ! I have the same issue, did you manage to fix it ?

**Comment by johnryt** [comment]
I'm also having this problem and getting this error, though just binary classification, and it's happening in the cross-validation step of the ```randomized_search``` function, implemented like this:
```
cat_indices = np.where(X_train[features_to_use].dtypes!=float)[0]
model = CatBoostClassifier(class_weights=class_weight, random_state=42, cat_features=cat_indices, silent=True)
_ = model.randomized_search(search_params, cv=5, n_iter=n_iter, verbose=True, X=X_train[features_to_use], y=y_train, plot=False)
```

My colleague implemented it differently, but got the same error (removing 'class_weights':class_weight from the params.update part removes the error. Same with removing class_weights=class_weight from the CatBoostClassifier initialization above):
``` 
from sklearn.model_selection import ParameterSampler
from catboost import CatBoostClassifier, Pool, cv

cat_feature_indices = np.where(X_train[features_to_use].dtypes != float)[0].tolist()
train_pool = Pool(X_train[features_to_use], y_train, cat_features=cat_feature_indices)

param_list = list(ParameterSampler(search_params, n_iter=20, random_state=42))

best_score = -np.inf
best_params = None

for params in param_list:
    params.update({
        'iterations': 300,
        'loss_function': 'Logloss',
        'task_type': 'CPU',
        'verbose': False,
        'class_weights': class_weight,
    })

    try:
        cv_result = cv(
            params=params,
            pool=train_pool,
            fold_count=5,
            partition_random_seed=42,
            stratified=True,
            early_stopping_rounds=20,
            verbose=False
        )
        score = -np.min(cv_result['test-Logloss-mean']) 
        if score > best_score:
            best_score = score
            best_params = params
model = CatBoostClassifier(**best_params)
model.fit(train_pool)
```
